util/dnsname/dnsname.go

```diff
@@ -41,18 +39,15 @@ func ToFQDN(s string) (FQDN, error) {
 
 	st := 0
 	for i := 0; i < len(s); i++ {
+		// Ignore leading '.' from wildcards for label processing
+		if i == 0 && s[i] == '.' {
+			st = i + 1
+			continue
+		}
 		if s[i] != '.' {
 			continue
 		}
 		label := s[st:i]
-		// You might be tempted to do further validation of the
```

Comment:
I'm not sure if this was intentional, we should keep some amount of it at least, it's reasonable advice.